js/index.js

```diff
@@ -1,6 +1,107 @@
+console.log("I'm ready!")
+
 // Iteration 1: Names and Input
-//
+
+const hacker1 = 'Paloma'
+console.log(`The driver´s name is ${hacker1}`)
+const hacker2 = 'Sergio'
+console.log(`The driver´s name is ${hacker2}`)
+
 // Iteration 2: Conditionals
 
+if (hacker1.length > hacker2.length) {
+    console.log(`The driver has the longest name, it has ${hacker1.length} characters `)
+} else if (hacker1.length < hacker2.length) {
+    console.log(`It seems that the navigator has the longest name, it has ${hacker2.length} characters.`)
+} else {
+    console.log(`Wow, you both have equally long names, ${hacker1.length} characters!.`)
+}
+
 
 // Iteration 3: Loops
+
+let hacker1Spacing = ''
+for (let i = 0; i < hacker1.length; i++) {
+    hacker1Spacing += hacker1[i] + ' ';
+}
+console.log(hacker1Spacing.toUpperCase().slice(0,-1));
+
+let hacker2Reverse = ''
+for (let i = hacker2.length - 1; i >= 0; i--) {
+    hacker2Reverse += hacker2[i];
+}
+console.log(hacker2Reverse);
+
+// Option 1
+for (let i = 0; i < hacker1.length; i++) {
+    if (hacker1[i] < hacker2[i]) {
+        console.log(`The driver's name goes first.`);
+        break;
+    } else if (hacker1[i] > hacker2[i]) {
+        console.log(`Yo, the navigator goes first definitely`);
+        break;
+    } else {
```

Comment:
Wow!!! genial!! pero justo aqui tenemos un buguillo, a ver, cuando entra por primera vez en el FOR, va a evaluar el IF no?, bien, si entra perfecto, si no para al else if, si entra perfecto!, peeeeero si pasa por el else necesariamente no tiene que enseñar el mensaje y sobre todo no tiene que hacer el break, me explico, si tenemos dos nombres: Paco y Pepe, segun tu codigo diria que son el mismo y se iria, y eso pues no es asi, tendria que seguir iterando, te voy a dar una idea y me dices si te parece bien, usando un booleano en ese else, y preguntando fuera del for por el, si no lo ves claro lo vemos vale?

Reply:
Algo así??: Seguramente algo más corto no? Es que me cuesta trabajar con los booleanos jajaja

let isTheSameName = true;
for (let i = 0; i < hacker1.length; i++) {
    if (hacker1[i] < hacker2[i]) {
        console.log(`The driver's name goes first.`);
        isTheSameName = false;
    } else if (hacker1[i] > hacker2[i]) {
        console.log(`Yo, the navigator goes first definitely`);
        isTheSameName = false;
    } else {
        isTheSameName = true;
    }
}
if (isTheSameName) {
    console.log(`What?! You both have the same name?`)